File: app/src/main/java/knf/kuma/tv/anime/RecentsPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.pojos.RecentObject;
import knf.kuma.tv.cards.RecentsCardView;
import knf.kuma.tv.details.TVAnimesDetails;

public class RecentsPresenter extends Presenter {

    public RecentsPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new RecentsCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((RecentsCardView) viewHolder.view).bind((RecentObject) item);
        viewHolder.view.setOnLongClickListener(v -> {
            AnimeObject object = CacheDB.INSTANCE.animeDAO().getByAid(((RecentObject) item).aid);
            if (object != null)
                TVAnimesDetails.start(v.getContext(), object.link);
            return true;
        });
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/RecentsPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import knf.kuma.pojos.av1.RecentAV1
import knf.kuma.tv.cards.RecentsCardView
import knf.kuma.tv.details.TVAnimesDetails

class RecentsPresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(RecentsCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any?) {
        if (item == null || item !is RecentAV1) return
        (viewHolder.view as RecentsCardView).bind(item)
        viewHolder.view.setOnLongClickListener { v ->
            TVAnimesDetails.start(v.context, item.animeUrl)
            true
        }
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
